app/src/main/java/fe/linksheet/activity/SelectDomainsConfirmationActivity.kt

```diff
@@ -0,0 +1,147 @@
+                                            repository.insert(domains.map {
+                                                PreferredApp(
+                                                    host = it,
+                                                    packageName = callingPackage,
+                                                    component = callingComponent,
+                                                    alwaysPreferred = true,
+                                                )
+                                            })
+                                        }
+
+                                        finish()
+                                    }
+                                },
+                                dynamicHeight = true,
+                                notifyCloseNoState = true,
+                            ) { state, close ->
+                                DialogColumn {
+                                    HeadlineText(headlineId = R.string.domain_selection_confirmation_title)
+                                    SubtitleText(subtitle = stringResource(id = R.string.domain_selection_confirmation_subtitle, appLabel))
+                                    DialogSpacer()
+                                    DialogContent(
+                                        items = state,
+                                        key = { it },
+                                        bottomRow = {
+                                            TextButton(
+                                                onClick = {
+                                                    close(false)
+                                                },
+                                            ) {
+                                                Text(text = stringResource(id = R.string.no))
+                                            }
+
+                                            TextButton(
+                                                onClick = {
+                                                    close(true)
+                                                },
+                                                colors = ButtonDefaults.textButtonColors(
+                                                    contentColor = MaterialTheme.colorScheme.error,
+                                                ),
+                                            ) {
+                                                Text(text = stringResource(id = R.string.yes))
+                                            }
+                                        },
+                                        content = {
+                                            Text(text = it)
+                                        },
+                                    )
+                                }
+                            }
+
+                            LaunchedEffect(null) {
+                                dialog.open()
+                            }
```

Comment:
Not entirely sure why, but just immediately calling `dialog.open()` doesn't work. The effect is needed.

Reply:
Yeah, this is probably a Compose thing. Haven't experienced it personally, as I have only opened dialogs on button click. Should be fine.